src/main/kotlin/me/ezra_home/retail_software_solution/rest/endpoints/sysuser/FruitEndpoint.kt

```diff
@@ -0,0 +1,78 @@
+package me.ezra_home.retail_software_solution.rest
+
+import me.ezra_home.retail_software_solution.businesses.sysuser.service.FruitService
+import me.ezra_home.retail_software_solution.model.entity.FruitEntity
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.util.UUID
+
+@RestController
+@RequestMapping("/api/fruits")
+class FruitController(
+    private val fruitService: FruitService
+) {
+
+    /**
+     * Get all fruits created by a specific user.
+     * @param userId UUID of the user.
+     * @return A list of FruitEntity.
+     */
+    @GetMapping("/user/{userId}")
+    fun getFruitsByUser(@PathVariable userId: UUID): ResponseEntity<List<FruitEntity>> {
```

Comment:
You should only expose DTOs/Beans to an API. You should never access your entity class anywhere other than the repository.

For this example, you are free to use entity class in the services, but that will change after I show you how to use caches. 

DTO get map to an entity just before save/update